docs/upgrade-guide.md

```diff
@@ -2,23 +2,54 @@
+
+Quick summary:
+
+- All direct references to `gl: WebGLRenderingContext` should be replaced with `device: Device` references. `impoert {Device} from '@luma.gl/core'`.
+- GPU resource classes should no longer be created directly. Use `device.createBuffer()`, `device.createTexture()` etc.
+- luma.gl no longer uses `GL` constants for GPU parameters, but instead uses (WebGPU style) string constants. Applications should no longer import `@luma.gl/constants`.
+
+For details, refer to the [luma.gl v9 Upgrade Guide](https://luma.gl/docs/upgrade-guide)
+
+**Deck**
+
+| v8 API                          | v9 API                        | Description |
+| ------------------------------- | ----------------------------- | ----------- |
+| `props.onWebGLInitialized(...)` | `props.onDeviceInitialized()` |             |
+
+
+**Layer**
+
+| v8 API                             | v9 API                                 | Description               |
+| ---------------------------------- | -------------------------------------- | ------------------------- |
+| `props.parameters: {GL constants}` | `props.parameters({string constants})` | Use new string constants. |
+
+
+**Custom Layers**
+
+Model creation needs to adapt to the [luma.gl v9 API](https://luma.gl/docs/upgrade-guide).
+
+| v8 API                               | v9 API                                   | Description                                   |
+| ------------------------------------ | ---------------------------------------- | --------------------------------------------- |
+| `new Model(this.context.gl, {opts})` | `new Model(this.context.device, {opts})` | Create models with `device`` instead of `gl`. |
+
+**Custom Shaders**
 
+- deck.gl no longer supports GLSL 1.00 shaders. All shaders must be updated to GLSL 3.00.
 
-- drawModes `GL.TRIANGLE_FAN` and `GL.LINE_LOOP` are not supported on WebGPU. Select different topology when creating geometries.
+**WebGPU preparation**
 
-### Deck
+While the 9.0 release of deck.gl does not yet support WebGPU, our goal is to enable WebGPU soon in a 9.x release. A number of changes will be required to deck.gl applications beyond the upgrades listed above. Again, these changes mostly affect customer layers.
 
-- `Deck.pickObjects` minor breaking change: if the same `data` is used by multiple top-level layers (e.g. a `ScatterplotLayer` and a `TextLayer`) that are visible in the picking bounds, `pickObjects` will yield one result for each picked object+layer combination, instead of one result for each picked object in previous versions.
+- deck.gl now uses uniform buffers instead of global uniforms. It is not yet required to use uniform buffers but it will be necessary if you would like to run deck.gl on WebGPU in future releases.
+- drawModes `GL.TRIANGLE_FAN` and `GL.LINE_LOOP` are not supported on WebGPU. Select a different topology when creating geometries.
```

Comment:
```suggestion
- drawModes `GL.TRIANGLE_FAN` and `GL.LINE_LOOP` are not supported on WebGPU. Select a different topology when creating geometries.
- The luma picking module now [uses uniform buffers](https://github.com/visgl/luma.gl/blob/master/modules/shadertools/src/modules/engine/picking/picking.ts#L34-L50). To access picking state in shaders use `picking.isActive` rather than `picking_isActive`
```